src/s3/requests.c

```diff
@@ -498,11 +503,17 @@ s3_put_object_with_contents(char *objectname, Pointer data, uint64 dataSize,
 	char	   *tmp;
 	int			sc;
 	StringInfoData buf;
-	unsigned char hash[32];
 	long		http_code = 0;
 
-	(void) SHA256((unsigned char *) data, dataSize, hash);
-	contenthash = hex_string((Pointer) hash, sizeof(hash));
+	if (dataHash == NULL)
+	{
+		unsigned char hash[PG_SHA256_DIGEST_STRING_LENGTH];
```

Comment:
Should be PG_SHA256_DIGEST_LENGTH not PG_SHA256_DIGEST_STRING_LENGTH
This can influence hex_string() as this size is transferred to it. (Maybe it'll fix valgrind test)

Reply:
I replaced it by `SHA256_DIGEST_LENGTH` here and other places where `PG_SHA256_DIGEST_STRING_LENGTH` was used mistakenly.
I added `O_SHA256_DIGEST_STRING_LENGTH` to remove dependency from `common/sha2.h`.